src/tools/cephfs/top/cephfs-top

```diff
@@ -644,116 +590,252 @@ class FSTop(object):
                      "kernel_version" in metadata])
                 num_libs = num_clients - (num_mounts + num_kclients)
         now = datetime.now().ctime()
-        self.header.addnstr(0, 0, FS_TOP_VERSION_HEADER_FMT.format(
-                            prog_name=FS_TOP_PROG_STR, now=now),
-                            hlen, curses.A_STANDOUT | curses.A_BOLD)
-        self.header.addnstr(1, 0, FS_TOP_CLIENT_HEADER_FMT.format(
-                            num_clients=num_clients, num_mounts=num_mounts,
-                            num_kclients=num_kclients, num_libs=num_libs), hlen)
-        self.header.addnstr(2, 0, headline, hlen)
-        self.header.refresh()
+        self.header.addstr(0, 0, FS_TOP_VERSION_HEADER_FMT.format(prog_name=FS_TOP_PROG_STR,
+                                                                  now=now), curses.A_BOLD)
+        self.header.addstr(2, 0, screen_title, curses.color_pair(color_id) | curses.A_BOLD)
+        self.header.addstr(3, 0, FS_TOP_CLIENT_HEADER_FMT.format(num_clients=num_clients,
```

Comment:
@vshankar IIRC, you suggested to have the fields name along with the filesystem name line should be fixed while scrolling so that user can have the glance of which filesystem metrics he's looking currently and can look the fields easily.If so, then currently we are not having the filesystem names lines fixed on scrolling.See,
Otherwise, rest looks good to me.

Reply:
Wasn't that related to @joscollin changes that introduce scrolling/pagination?

Reply:
I am talking about the suggestion in the watercooler for scrolling, if not wrong.

Reply:
There are no qt kind of widgets in curses to achieve such things. The text displayed in the fsstats pad is pre-loadad as a whole. The current text should be erased in-order to display the following text at the same position while scrolling. In other words, each scroll key press refreshes different portion of the pad (which is already text loaded), which creates the illusion of scrolling. So there's no way to stop the movement of `Filesystem;` name line, which is different for each fs object in the code. In other words, curses doesn't identify it as a special line (or object) to impose some restrictions to it. The top line moving was stopped, as it's common text for all filesystems. So I've drawn it only at the top and refreshing from the next line onwards. Hope that clarifies things.

Reply:
> @vshankar IIRC, you suggested to have the fields name along with the filesystem name line should be fixed while scrolling so that user can have the glance of which filesystem metrics he's looking currently and can look the fields easily.If so, then currently we are not having the filesystem names lines fixed on scrolling.See, [5493030](https://github.com/ceph/ceph/commit/549303071d7d8b77abeae856cfb89a2f7378a711) Otherwise, rest looks good to me.

OK. What do you mean by "not having filesystem names lines fixed on scrolling"?

Reply:
He meant something shown in this video: https://www.youtube.com/watch?v=6zS0OavqbAc. Watch at 0:40.

Reply:
> > @vshankar IIRC, you suggested to have the fields name along with the filesystem name line should be fixed while scrolling so that user can have the glance of which filesystem metrics he's looking currently and can look the fields easily.If so, then currently we are not having the filesystem names lines fixed on scrolling.See, [5493030](https://github.com/ceph/ceph/commit/549303071d7d8b77abeae856cfb89a2f7378a711) Otherwise, rest looks good to me.
> 
> OK. What do you mean by "not having filesystem names lines fixed on scrolling"?

I meant to say if we scroll now, the `TOP_LINE` (the line where the names of fields are displayed ) is always visible but the line `Filesystem: cephfs - 1 client(s)` is being scrolled along with the metrics too.
As per @joscollin reply and my findings, I can say its not easy to have that in ncurses. But I am unsure whether we can't implement. But that's only w.r.t the suggestion, it looks fine even without that.

Reply:
@neesingh-rh The content changes are not within the scope of this PR. Only scrolling fixes are in scope. But I've still incorporated the sensible things mentioned in the WC, especially making the table header static and it works very well.